cbdr/cortex
# CE-173: Rails 5.1 Upgrade

## Purpose:
This PR accomplishes the following:
* Finalizes upgrade from Rails `5.0` -> `5.1` (which was partially complete)
* Ports over all Rails 5 config changes/improvements, comments, binstubs, etc
* Upgrades Ruby to `v2.4.2`
* Removes Bower, replaces with Yarn; moves all client-side dependencies to Yarn
* New Sprockets `4.x` config
* Improves sourcecode change detection/auto-reload
* Upgrades Gemfile
* Upgrades C66 server dependencies
* Updates documentation

## JIRA:
https://careerbuilder.atlassian.net/browse/CE-173

## Steps to Take On Prod
N/A

## Changes:
* Changes to setup
  * N/A

* Architectural changes
  * N/A

* Migrations
  * N/A

* Library changes
  * `cortex-plugins-core` upgrade required

* Side effects
  * N/A

## Screenshots
* Before
N/A

* After
N/A

## QA Links:
http://web.cortex-2.development.c66.me/

## How to Verify These Changes
* Specific pages to visit
  * All

* Steps to take
  * Login
  * Ensure tenant switcher works
  * Ensure Blog `ContentItems` can be created and updated

* Responsive considerations
  * N/A

## Relevant PRs/Dependencies:
https://github.com/cortex-cms/cortex-plugins-core/pull/65

## Additional Information
N/A
